numpy/core/tests/test_scalarmath.py

```diff
@@ -189,30 +189,31 @@ def test_modular_power(self):
             assert_raises(TypeError, operator.pow, np.array(t(a)), b, c)
 
 
-class TestModulus(TestCase):
+def floordiv_and_mod(x, y):
+    return (x // y, x % y)
+
 
-    floordiv = operator.floordiv
-    mod = operator.mod
+class TestModulus(TestCase):
 
     def test_modulus_basic(self):
         dt = np.typecodes['AllInteger'] + np.typecodes['Float']
-        for dt1, dt2 in itertools.product(dt, dt):
-            for sg1, sg2 in itertools.product((+1, -1), (+1, -1)):
-                if sg1 == -1 and dt1 in np.typecodes['UnsignedInteger']:
-                    continue
-                if sg2 == -1 and dt2 in np.typecodes['UnsignedInteger']:
-                    continue
-                fmt = 'dt1: %s, dt2: %s, sg1: %s, sg2: %s'
-                msg = fmt % (dt1, dt2, sg1, sg2)
-                a = np.array(sg1*71, dtype=dt1)[()]
-                b = np.array(sg2*19, dtype=dt2)[()]
-                div = self.floordiv(a, b)
-                rem = self.mod(a, b)
-                assert_equal(div*b + rem, a, err_msg=msg)
-                if sg2 == -1:
-                    assert_(b < rem <= 0, msg)
-                else:
-                    assert_(b > rem >= 0, msg)
+        for op in [floordiv_and_mod, divmod]:
```

Comment:
Perhaps before this loop, or even at the file level:
```
signs = {
    d: (+1,) if dt in np.typecodes['UnsignedInteger'] else (+1, -1)
    for d in dt
}
```

And then `itertools.product(signs[dt1], signs[dt2])` below, which removes the `continue`s